crates/fuel-core/src/graphql_api.rs

```diff
@@ -87,14 +87,14 @@ pub const DEFAULT_QUERY_COSTS: Costs = Costs {
     da_compressed_block_read: 4000,
 };
 
-pub fn query_costs() -> &'static Costs {
-    QUERY_COSTS.get().unwrap_or(&DEFAULT_QUERY_COSTS)
+pub fn query_costs(extract_field: impl FnOnce(&Costs) -> usize) -> usize {
+    extract_field(&QUERY_COSTS.read().unwrap())
```

Comment:
I don't think we should sunset the `OnceLock` usage, because in regular node operation, we do not expect these values to change once the node has started. either hide it behind a `RwLock` only in the `test-helpers` feature, and if not, it should still use `OnceLock`. you can also force initialization.

- https://github.com/FuelLabs/fuel-core/blob/a952ff44e0c73323f87f0669112f4a76db6dfc61/benches/src/utils.rs#L150 we run these sequentially to prevent env conflicts for example
- https://doc.rust-lang.org/beta/std/sync/struct.OnceLock.html#method.get_mut get_mut

at the start of the test, mutably alter the QUERY_COSTS, and then after set it back to its initial value, this *might* work ?

Reply:
We've synced on this and agreed to:

1. Use the `OnceLock` solution.
2. Disallow tests from initializing non-default values.

I've implemented this in https://github.com/FuelLabs/fuel-core/pull/2335/commits/8f93f4f85dcb67774b061d107dd591e2164b5d5a now.